app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextModuleActions.java

```diff
@@ -381,4 +413,58 @@ public void onClick(DialogInterface dialog, int id) {
 
         builder.show();
     }
+
+    /**
+     * Show the camera picker via intent
+     * Source: http://developer.android.com/training/camera/photobasics.html
+     */
+    private void showCameraDialog() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(_activity.getPackageManager()) != null) {
+            File photoFile = null;
+            try {
+                // Create an image file name
+                String imageFileName = _activity.getString(R.string.app_name) + "_" + System.currentTimeMillis();
+                File storageDir = new File(Environment.getExternalStoragePublicDirectory(
+                        Environment.DIRECTORY_DCIM), "Camera");
+                photoFile = File.createTempFile(imageFileName, ".jpg", storageDir);
+
+                // Save a file: path for use with ACTION_VIEW intents
+                _cameraPictureFilepath = photoFile.getAbsolutePath();
+
+            } catch (IOException ex) {
+                _au.showSnackBar(R.string.main__error_camera_cannot_start, false);
+            }
+
+            // Continue only if the File was successfully created
+            if (photoFile != null) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+//                    Log.d("Markdown", _cameraPictureFilepath);
+//                    Log.d("Markdown", _activity.getString(R.string.app_fileprovider));
+                    Uri uri = FileProvider.getUriForFile(_activity,
+                            _activity.getString(R.string.app_fileprovider), photoFile);
+                    takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, uri);
+                } else {
+                    takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+                }
+                _au.animateToActivity(takePictureIntent, false, REQUEST_TAKE_CAMERA_PICTURE);
+            }
+        }
+    }
+
+    private void onPictureTaken(String imagePath) {
+        String url = imagePath.replace(")", "\\)")
+                .replace(" ", "%20");  // Workaround for parser - cannot deal with spaces and have other entities problems
+
+        String formatTemplate = "![%s](%s)";
```

Comment:
If it doesnt work like you mentioned - I would start with just doing `insert` at `getSelectionStart`. If that and only that works, it is also fine. We can polish it in a further PR if you want to.